test/util/schemaform-utils.jsx

```diff
@@ -39,8 +43,11 @@ export class DefinitionTester extends React.Component {
   handleChange = (data) => {
     const state = this.props.state;
     const uiSchema = this.state.uiSchema;
-    const formData = _.merge(this.props.formData, data);
-    let schema = updateRequiredFields(this.state.schema, uiSchema, formData);
+    const formData = this.state.schema.type === 'array'
+      ? data
+      : _.merge(this.props.formData, data);
+    let schema = updateItemsSchema(this.state.schema, formData);
+    schema = updateRequiredFields(schema, uiSchema, formData);
```

Comment:
It might be good to wrap this whole sequence of function calls in a function that we could reuse more easily.